src/deploymentstatus.h

```diff
@@ -6,25 +6,44 @@
 #define BITCOIN_DEPLOYMENTSTATUS_H
 
 #include <chain.h>
+#include <versionbits.h>
 
 #include <limits>
 
+/** Global cache for versionbits deployment status */
+extern VersionBitsCache versionbitscache;
+
 /** Determine if a deployment is active for the next block */
 inline bool DeploymentActiveAfter(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::BuriedDeployment dep)
 {
     return (pindexPrev == nullptr ? 0 : pindexPrev->nHeight + 1) >= params.DeploymentHeight(dep);
 }
 
+inline bool DeploymentActiveAfter(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::DeploymentPos dep)
+{
+    return ThresholdState::ACTIVE == VersionBitsState(pindexPrev, params, dep, versionbitscache);
```

Comment:
In commit "[refactor] Add versionbits deployments to deploymentstatus.h":

Perhaps assert that dep >= 0 in these functions (and dep < 0 for the buried ones)?

Reply:
I've added a `ValidDeployment()` function for checking this with a bit more precision.

Might be worth revisiting @JeremyRubin's suggestion of using templated functions so it can be a static assertion that's resolved at compile-time? Having looked at it again, I think the only thing that would really end up making awkward is the rpc/blockchain.cpp stuff (ie, it would mean templating the `SoftForkPushBack` function and having the compiler duplicate the code and possibly not de-duplicate it when optimising) but that's not necessarily a big deal.

I've added a draft commit that switches to templated functions and static_asserts to see what that's like. It does make rpc/blockchain a bit bigger (after compiling with clang) so might need be worth some tweaking to avoid that, and the syntax is a bit clunky, but it seems plausible. Thoughts?

Reply:
Okay, dropped the templated approach as too unwieldy for not enough benefit, so think this is resolved via `ValidDeployment()`.